### DISCUSSION TITLE
Insert Text Heading Based on File Extension Given

### DISCUSSION THREAD

**Discussion by rheniser** [microsoft/vscode-discussions#467] [Extension Development QnA] [answered]
Is there a way (using built-in commands or an extension) to insert text (a header) into a newly created file when it is added based on the file extension given? 

I am thinking about creating an extension to do this. I just wanted to make sure I am not reinventing the wheel.

Kind regards,
Ryan

**Comment by alefragnani** [comment, marked as answer]
Hi @rheniser ,

I guess this extension, called [Power Header](https://marketplace.visualstudio.com/items?itemName=epivision.vscode-file-header), does that.

I’ve never used it, but based on its readme, it seems to support the newly created scenario you described.

Hope this helps

**Comment by rheniser** [comment]
Thank you!
